hack/generator/pkg/astmodel/armconversion/convert_to_arm_function_builder.go

```diff
@@ -41,15 +42,16 @@ func newConvertToARMFunctionBuilder(
 			isSpecType:            c.isSpecType,
 			codeGenerationContext: codeGenerationContext,
 		},
-		resultIdent: "result",
+		resultIdent:           "result",
+		typeConversionBuilder: astmodel.NewConversionFunctionBuilder(c.idFactory, codeGenerationContext),
 	}
 
+	result.typeConversionBuilder.AddConversionHandlers(result.convertComplexTypeNameProperty)
 	result.propertyConversionHandlers = []propertyConversionHandler{
 		result.namePropertyHandler,
 		result.fixedValuePropertyHandler("Type"),
 		result.fixedValuePropertyHandler("ApiVersion"),
-		result.propertiesWithSameNameAndTypeHandler,
-		result.propertiesWithSameNameButDifferentTypeHandler,
```

Comment:
Because of the `definedErrVar` which I'm currently using to stick a `var err` at the top of functions that need it.
We need this because the following isn't valid code:
```
err := populateFromARM(foo)
err := populateFromARM(bar)
```

The current way I have things is kinda icky though because it means that these top level handlers need to understand a bit about what the lower level handler is going to do in order to determine if it needs to start with that `var err` or not. I think I see a way to fix this and make the `from` handler the same as the `to` handler... will investigate

Reply:
I have a fix for this - it's with a bunch of other renaming stuff though so will include it in a second PR I think